types/wlr_screencopy_v1.c

```diff
@@ -434,6 +492,15 @@ static void capture_output(struct wl_client *wl_client,
 
 	zwlr_screencopy_frame_v1_send_buffer(frame->resource, frame->format,
 		buffer_box.width, buffer_box.height, frame->stride);
+
+	if (version >= 3) {
+		uint32_t fourcc = fourcc_from_wl_shm_format(frame->format);
```

Comment:
The preferred OpenGL read format isn't the same as the format used by the output buffer.

Reply:
But `wlr_output_preferred_read_format()` yields `enum wl_shm_format`

Reply:
`wlr_output_preferred_read_format` is only applicable to wl_shm, not to DMA-BUFs.

Reply:
This is why I convert the wl_shm format to fourcc. Do you mean that the conversion fourcc -> gl read format -> wl_shm_format -> fourcc might not be accurate? Does there exist some way to read the fourcc format of the dmabuf directly?

Reply:
What I mean is that the format returned by `wlr_output_preferred_read_format` will not necessarily be the same as the one used by `wlr_output_export_dmabuf` (modulo conversion). `wlr_output_preferred_read_format` is about `wlr_renderer_read_pixels` only, nothing else.

Reply:
Oh, right, I found it. It's in the `struct wlr_dmabuf_attributes`.